src/Compiler/Microsoft.AspNetCore.Razor.Language/test/Syntax/FindTokenTests.cs

```diff
@@ -0,0 +1,883 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Language.Test;
+
+public class FindTokenTests
+{
+    private const string PositionMarker = "$$";
+
+    private static (RazorSyntaxTree Tree, int Position) ParseWithPosition(string textWithPosition)
```

Comment:
We do have access to the `TestFileMarkupParser` that does this, though I admit this method is not exactly complicated. You could use it and change the tests to do `<$${|Text:div|}></div>` though, if you like brevity. Then again with a target of 1000 lines maybe not 😛

Reply:
I don't want spans, I want positions 🙂